extension/scripts/features/abroad-items-filter/ttAbroadItemsFilter.js

```diff
@@ -0,0 +1,153 @@
+"use strict";
+
+(async () => {
+	if (!getPageStatus().access) return;
+
+	featureManager.registerFeature(
+		"Item Filter",
+		"travel",
+		() => settings.pages.travel.itemFilter,
+		null,
+		addFilter,
+		removeFilter,
+		{
+			storage: ["settings.pages.travel.itemFilter"],
+		},
+		null
+	);
+
+	function addFilter() {
+		const { content } = createContainer("Item Filters", {
+			nextElement: document.find(".travel-agency-market"),
+		});
+		// Create all checkboxes beforehand
+		const cbProfitOnly = createCheckbox("Only Profit");
+		cbProfitOnly.onChange(applyFilters);
+		const profitOnlyFilter = document.newElement({
+			type: "div",
+			class: "profitOnlyFilter",
+			children: [
+				document.newElement({ type: "strong", text: "Profit" }),
+				cbProfitOnly.element,
+			],
+		});
+		const cbsCategories = createCheckboxList(
+			[
+				{ id: "plushie", description: "Plushies" },
+				{ id: "flower", description: "Flowers" },
+				{ id: "drug", description: "Drugs" },
+				{ id: "weapon", description: "Weapons" },
+				{ id: "armor", description: "Armor" },
+				{ id: "other", description: "Other" },
+			],
+			"column"
+		);
+		cbsCategories.onSelectionChange(applyFilters);
+		const categoryFilter = document.newElement({
+			type: "div",
+			class: "categoryFilter",
+			children: [
+				document.newElement({ type: "strong", text: "Categories" }),
+				cbsCategories.element,
+			],
+		});
+		// Append them ALL
+		content.appendChild(document.newElement({
+			type: "div",
+			class: "statistics",
+			children: [
+				"Showing ",
+				document.newElement({ type: "strong", class: "count", text: "X" }),
+				" of ",
+				document.newElement({ type: "strong", class: "total", text: "Y" }),
+				" items",
+			],
+		}));
+		content.appendChild(document.newElement({
+			type: "div",
+			class: "content",
+			children: [
+				profitOnlyFilter,
+				categoryFilter
+			],
+		}));
+
+		cbProfitOnly.setChecked(filters.abroadItems.profitOnly);
+		cbsCategories.setSelections(filters.abroadItems.categories);
+
+		applyFilters();
+
+		async function applyFilters() {
+			const profitOnly = settings.pages.travel.travelProfits && cbProfitOnly.isChecked();
+			const categories = cbsCategories.getSelections();
+			const categoriesExtra = [];
+
+			// Categories
+			for (const category of categories) {
```

Comment:
Category checking should work better than this. Preferably as in the travel table: check the item category, if it's primary or melee it's a weapon, else it's 'other'. This way, even if new categories are added for items, it won't break.